git-branchless-revset/src/builtins.rs

```diff
@@ -60,6 +61,21 @@ lazy_static! {
     };
 }
 
+#[instrument]
+fn fn_abandoned(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
+    eval0(ctx, name, args)?;
+
+    let obsolete_commits = ctx.dag.query_obsolete_commits();
+    let abandoned_children = find_abandoned_children(
+        obsolete_commits,
+        // ?
+        // ?
+        // ?
+        // ?
```

Comment:
For these three parameters, you can copy the lines here:

https://github.com/arxanas/git-branchless/blob/c6292d28cbca375785720e46c86270fbbbccf7ff/git-branchless-revset/src/builtins.rs#L493-L500

Ideally, we would change it to lazily construct these objects, but that's harder to do in Rust.

Reply:
What is the use of the `oid` param? My guess was that it was the "base"(?) commit to start finding from but thought I would check before trying to supply it

Reply:
Oops, that's an important parameter to have forgotten to describe. It's probably best to check the source in case I misremember. I think it's the OID of the old base commit, which should have been rewritten by now, whose children are currently abandoned.

Reply:
So in this context (looking for all abandoned commits), should it be the root commit of the repo?

Reply:
This only checks the immediate children of the provided commit, not all descendants recursively. It would be best to follow the logic in `restack.rs`. Here, we use the list of all obsolete commits as the starting point:

https://github.com/arxanas/git-branchless/blob/ac8a1876cfd7c14ff4ecc99e3ca54d7edc989180/git-branchless/src/commands/restack.rs#L100